java/core/src/main/java/co/worklytics/psoxy/gateway/ConfigService.java

```diff
@@ -50,6 +53,30 @@ default boolean supportsWriting() {
      */
     void putConfigProperty(ConfigProperty property, String value);
 
+    /**
+     * write value of property in config, if supports it
+     *
+     * @param property to write value for
+     * @param value to write
+     * @throws WritePropertyRetriesExhaustedException if write fails after designated retries
+     */
+    default void putConfigProperty(ConfigProperty property, String value, int retries) throws WritePropertyRetriesExhaustedException {
+        if (retries <= 0) {
+            // use the non-retry version
+            throw new IllegalArgumentException("retries must be > 0");
+        }
+        do {
+            try {
+                putConfigProperty(property, value);
+                return;
+            } catch (Exception ignore) {
```

Comment:
add logging?

Reply:
:+1: passed through the cause exception. Can't use `@Log` because is an interface